FirebaseAuth/Sources/Swift/SystemService/AuthNotificationManager.swift

```diff
@@ -69,56 +68,53 @@
       self.application = application
       self.appCredentialManager = appCredentialManager
       timeout = kProbingTimeout
+      condition = AuthCondition()
     }
 
-    /// Checks whether or not remote notifications are being forwarded to this class.
-    /// - Parameter callback: The block to be called either immediately or in future once a result
-    /// is available.
-    func checkNotificationForwardingInternal(withCallback callback: @escaping (Bool) -> Void) {
-      if pendingCallbacks != nil {
-        pendingCallbacks?.append(callback)
-        return
+    private actor PendingCount {
+      private var count = 0
+      func increment() -> Int {
+        count = count + 1
+        return count
       }
+    }
+
+    private let pendingCount = PendingCount()
+
+    /// Checks whether or not remote notifications are being forwarded to this class.
+    func checkNotificationForwarding() async -> Bool {
       if let getValueFunc = immediateCallbackForTestFaking {
-        callback(getValueFunc())
-        return
+        return getValueFunc()
       }
       if hasCheckedNotificationForwarding {
-        callback(isNotificationBeingForwarded)
-        return
+        return isNotificationBeingForwarded
       }
-      hasCheckedNotificationForwarding = true
-      pendingCallbacks = [callback]
-
-      DispatchQueue.main.async {
-        let proberNotification = [self.kNotificationDataKey: [self.kNotificationProberKey:
-            "This fake notification should be forwarded to Firebase Auth."]]
-        if let delegate = self.application.delegate,
-           delegate
-           .responds(to: #selector(UIApplicationDelegate
-               .application(_:didReceiveRemoteNotification:fetchCompletionHandler:))) {
-          delegate.application?(self.application,
-                                didReceiveRemoteNotification: proberNotification) { _ in
+      if await pendingCount.increment() == 1 {
```

Comment:
IIUC, `pendingCount` is only used to enforce the following bracketed logic only gets called once? How was this enforced in the old diff?

Reply:
By either `_pendingCallbacks` being non-nil or `_hasCheckedNotificationForwarding` being `YES`.